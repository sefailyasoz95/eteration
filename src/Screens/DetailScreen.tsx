import { useNavigation, useRoute } from "@react-navigation/core";
import { NativeStackNavigationOptions, NativeStackNavigationProp } from "@react-navigation/native-stack";
import React, { useEffect, useState } from "react";
import { Image, StyleSheet, Text, View } from "react-native";
import { AppStackParamList } from "../StackParamLists/AppStackParamList";
import { RouteProp } from "@react-navigation/native";
import { imageUrlHelper } from "../Utils/Helpers/imageUrlHelper";

type Props = {
	navigation: NativeStackNavigationProp<AppStackParamList, "Detail">;
	route: RouteProp<AppStackParamList, "Detail">;
};
const DetailScreen = ({ navigation, route }: Props) => {
	const { simpson } = route.params;
	const options: NativeStackNavigationOptions = {
		headerTitle: `About ${simpson.name.split(" ")[0]}`,
	};
	const [isValidURL, setIsValidURL] = useState(false);

	useEffect(() => {
		setIsValidURL(imageUrlHelper(simpson?.avatar));
		navigation.setOptions(options);
	}, []);
	return (
		<View style={styles.container}>
			{isValidURL ? (
				<Image source={{ uri: simpson.avatar }} width={50} height={50} style={styles.image} />
			) : (
				<Text style={styles.noImage}>â›”</Text>
			)}
			<Text style={styles.name}>{simpson.name}</Text>
			<Text style={styles.job}>{simpson.job}</Text>
			<Text style={styles.about}>{simpson.about}</Text>
		</View>
	);
};

export default DetailScreen;

const styles = StyleSheet.create({
	container: {
		flex: 1,
		alignItems: "center",
		justifyContent: "flex-start",
		marginHorizontal: 13,
	},
	image: {
		height: 300,
		width: 170,
		resizeMode: "contain",
		marginTop: "2%",
	},
	noImage: {
		padding: 10,
	},
	name: {
		fontWeight: "bold",
		fontSize: 20,
		letterSpacing: 1,
		textTransform: "capitalize",
	},
	job: {
		fontSize: 16,
		marginTop: 5,
		textTransform: "capitalize",
		color: "rgba(0,0,0,0.8)",
	},
	about: {
		fontSize: 16,
		color: "#555",
		marginVertical: 15,
	},
});
