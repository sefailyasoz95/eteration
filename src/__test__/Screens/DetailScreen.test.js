import React from "react";
import { render } from "@testing-library/react-native";
import DetailScreen from "../../Screens/DetailScreen";

describe("init", () => {
	test("detail screen renders correctly", () => {
		const mockedParams = {
			route: { params: { simpson: { name: "simpson name" } } },
			navigation: { setOptions: ({}) => {} },
		};

		const tree = render(<DetailScreen {...mockedParams} />);
		expect(tree).toMatchSnapshot();
	});
});
