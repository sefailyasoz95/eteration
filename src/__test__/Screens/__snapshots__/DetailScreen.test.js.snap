// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`init detail screen renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "flex-start",
      "marginHorizontal": 13,
    }
  }
>
  <Text
    style={
      Object {
        "padding": 10,
      }
    }
  >
    â›”
  </Text>
  <Text
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
        "letterSpacing": 1,
        "textTransform": "capitalize",
      }
    }
  >
    simpson name
  </Text>
  <Text
    style={
      Object {
        "color": "rgba(0,0,0,0.8)",
        "fontSize": 16,
        "marginTop": 5,
        "textTransform": "capitalize",
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#555",
        "fontSize": 16,
        "marginVertical": 15,
      }
    }
  />
</View>
`;
